src/OpenTelemetry.Instrumentation.ElasticsearchClient/OpenTelemetry.Instrumentation.ElasticsearchClient.csproj

```diff
@@ -8,7 +8,8 @@
     <IncludeSharedInstrumentationSource>true</IncludeSharedInstrumentationSource>
   </PropertyGroup>
   <ItemGroup>
-    <PackageReference Include="OpenTelemetry" Version="1.2.0" />
+    <PackageReference Include="OpenTelemetry" Version="$(OpenTelemetryCoreLatestPrereleaseVersion)" />
+    <PackageReference Include="System.Reflection.Emit.Lightweight" Version="4.7.0" />
```

Comment:
I think that you should be able to avoid this reference.
`IncludeSharedInstrumentationSource` brings non-used implementation of classes which based on `DynamicMethod` (tests are using it, but the library does not need it).

Probably it will required a bit deeper cleanup, but it should be possible.